### PR TITLE
IntezerV2 Integration improvments

### PR DESCRIPTION
**Pull request by kfirstri**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Description
Updates to Intezer integration, including bug fixes, adjustments to best practices and addition of new command available to customers for interacting better with intezer analyze api.

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+url = "https://pypi.org/simple"
+verify_ssl = true
+name = "pypi"
+
+[packages]
+
+[dev-packages]
+
+[requires]
+python_version = "3.7"
```
Comment (yaakovpraisler) [inline Pipfile:11]: This Pipfile is out of the pack. Please remove it.

File: Pipfile.lock [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+            {
+                "name": "pypi",
+                "url": "https://pypi.org/simple",
+                "verify_ssl": true
+            }
+        ]
+    },
+    "default": {},
+    "develop": {}
+}
```
Comment (yaakovpraisler) [inline Pipfile.lock:20]: As above, this is out of the pack. Please remove it.

**Review by yaakovpraisler — COMMENTED** [review]
Looking good. Please see my comments -
Please update the README file with your new commands and outputs. Also please add unit tests.
Let's schedule a demo for the new commands.

File: Packs/Intezer/Integrations/IntezerV2/IntezerV2.yml [hunk 3]
```diff
@@ -3,157 +3,260 @@ commonfields:
...
+          description: status of the analysis
+          type: string
+        - contextPath: Intezer.Analysis.Type
+          description: type of the analysis
+          type: string
+    - name: intezer-get-analysis-result
+      arguments:
+        - default: false
+          description: The analysis ID we want to get results for
+          isArray: false
```
Comment (yaakovpraisler) [inline Packs/Intezer/Integrations/IntezerV2/IntezerV2.yml:111]: Why the change in this command?
It will break backward compatibility. 
The playbooks are using generic polling which uses this command as it, and this change will break it.

File: Packs/Intezer/Integrations/IntezerV2/IntezerV2.py [hunk 4]
```diff
@@ -38,164 +49,261 @@
...
+        'Intezer.Analysis(val.ID && val.ID == obj.ID)': {
+            'ID': analysis_id,
+            'Status': 'InProgress'
+        }
+    }
+
+    return CommandResults(
+        readable_output='Analysis is still in progress',
+        outputs=context_json
+    )
```
[the remark this answers is not in the document]
Comment (kfirstri) [reply Packs/Intezer/Integrations/IntezerV2/IntezerV2.py:82]: awesome 👍

File: Packs/Intezer/Integrations/IntezerV2/IntezerV2.yml (hunk 3 above)
Comment (kfirstri) [reply Packs/Intezer/Integrations/IntezerV2/IntezerV2.yml:111]: i see.. so i have to be able to handle arrays, do i have to return different CommandResult for each analysis id or just 1 with entry in the context for each? @yaakovpraisler

Comment (kfirstri) [reply Packs/Intezer/Integrations/IntezerV2/IntezerV2.yml]: ok nevermind, i return a list of command results for each analysis :)

Comment (yaakovpraisler) [reply Packs/Intezer/Integrations/IntezerV2/IntezerV2.yml]: If the argument continues to support comma-separated values, so you can also leave the outputs as they were until now.

File: Packs/Intezer/ReleaseNotes/1_2_0.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+#### Intezer v2
+
+- New Integration Param - `AnalyzeBaseURL` - Allowes setting the integration to work with a different Intezer Analyze instance (i.e on premise)
+- Modifying the integration to XSOAR integration development best practices
+- Fixes for the context not joining together
+- Integration now uses the Intezer Analyze Python SDK for future support and updates
+
+- Changes in command:
+    - `intezer-get-analysis-result` - From now recieves a single analysis id instead of many
```
Comment (yaakovpraisler) [inline Packs/Intezer/ReleaseNotes/1_2_0.md:11]: Remove this RN entry.

Comment (kfirstri) [reply Packs/Intezer/ReleaseNotes/1_2_0.md:11]: done

File: Packs/Intezer/Integrations/IntezerV2/IntezerV2.yml (hunk 3 above)
Comment (kfirstri) [reply Packs/Intezer/Integrations/IntezerV2/IntezerV2.yml]: it's not works as an array so it's good :+1:

File: Packs/Intezer/ReleaseNotes/1_2_0.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+- New Integration Param - `AnalyzeBaseURL` - Allowes setting the integration to work with a different Intezer Analyze instance (i.e on premise)
+- Modifying the integration to XSOAR integration development best practices
+- Fixes for the context not joining together
+- Integration now uses the Intezer Analyze Python SDK for future support and updates
+
+- New Commands - 
+    - `intezer-get-sub-analyses` - Get a list of the Analysis sub analyses
+    - `intezer-get-analysis-code-reuse` - Get the Code reuse information for an analyses
+    - `intezer-get-analysis-metadata` - Get an Analysis metdata (different file hashes and more)
+    - `intezer-get-family-info` - Get family information from Intezer Analyze
```
Comment (yaakovpraisler) [inline Packs/Intezer/ReleaseNotes/1_2_0.md:14]: Add here your new outputs.

File: Pipfile (hunk 1 above)
Comment (kfirstri) [reply Pipfile]: now removed

File: Pipfile.lock (hunk 2 above)
Comment (kfirstri) [reply Pipfile.lock]: now removed

File: README.md [hunk 7]
```diff
@@ -1,45 +0,0 @@
-![Content logo](xsoar_content_logo.png)
```
[the remark this answers is not in the document]
Comment (kfirstri) [reply README.md]: oops, it's back :+1:

File: Packs/Intezer/Integrations/IntezerV2/README.md [hunk 8]
```diff
@@ -1,322 +1,420 @@
...
+#### Command Example
+```
+intezer-get-latest-report file_hash="8cbf90aeab2c93b2819fcfd6262b2cdb"
+```
+
+#### Human Readable Output
+```
+Intezer File analysis result
+----
+SHA256: bf293bda73c5b4c1ec66561ad20d7e2bc6692d051282d35ce8b7b7020c753467
```
[the remark this answers is not in the document]
Comment (kfirstri) [reply Packs/Intezer/Integrations/IntezerV2/README.md:95]: it wasn't but i just remove all of them anyway

File: Packs/Intezer/ReleaseNotes/1_2_0.md (hunk 6 above)
Comment (kfirstri) [reply Packs/Intezer/ReleaseNotes/1_2_0.md:14]: Do you have an example? non of the last change logs contained "outputs"

File: Packs/Intezer/Integrations/IntezerV2/README.md [hunk 9]
```diff
@@ -1,322 +1,420 @@
...
+| Intezer.Analysis.ID | string | Intezer analysis id | 
+| Intezer.Analysis.Status | string | status of the analysis | 
+| Intezer.Analysis.Type | string | type of the analysis | 
+
+
+#### Command Example
+``` 
+!intezer-analyze-by-hash file_hash="8cbf90aeab2c93b2819fcfd6262b2cdb"
+```
+
```
[the remark this answers is not in the document]
Comment (kfirstri) [reply Packs/Intezer/Integrations/IntezerV2/README.md:61]: done

**Comment by kfirstri** [comment]
Hi @yaakovpraisler i just pushed more tests and changed, letting it run and check :)
when do you want to schedule a demo?

File: Packs/Intezer/Integrations/IntezerV2/IntezerV2.py [hunk 10]
```diff
@@ -1,22 +1,34 @@
...
 requests.packages.urllib3.disable_warnings()
 
-API_KEY = demisto.getParam('APIKey')
-SERVER_URL = 'https://analyze.intezer.com/api'
-API_VERSION = '/v2-0'
-BASE_URL = SERVER_URL + API_VERSION
 IS_AVAILABLE_URL = 'is-available'
 ERROR_PREFIX = 'Error from Intezer:'
 ACCEPTABLE_HTTP_CODES = {200, 201, 202}
-USE_SSL = not demisto.params().get('insecure', False)
```
Comment (yaakovpraisler) [inline Packs/Intezer/Integrations/IntezerV2/IntezerV2.py:19]: Please handle the ssl param, and also please if you can do it in the main function.

File: Packs/Intezer/Integrations/IntezerV2/IntezerV2_test.py [hunk 11]
```diff
@@ -0,0 +1,642 @@
...
+
+# from Packs.Base.Scripts.CommonServerPython.CommonServerPython import outputPaths
+# from Packs.Intezer.Integrations.IntezerV2.IntezerV2 import analyze_by_hash_command
+# from Packs.Intezer.Integrations.IntezerV2.IntezerV2 import analyze_by_uploaded_file_command
+# from Packs.Intezer.Integrations.IntezerV2.IntezerV2 import check_analysis_status_and_get_results_command
+# from Packs.Intezer.Integrations.IntezerV2.IntezerV2 import get_analysis_code_reuse_command
+# from Packs.Intezer.Integrations.IntezerV2.IntezerV2 import get_analysis_metadata_command
+# from Packs.Intezer.Integrations.IntezerV2.IntezerV2 import get_analysis_sub_analyses_command
+# from Packs.Intezer.Integrations.IntezerV2.IntezerV2 import get_family_info_command
+# from Packs.Intezer.Integrations.IntezerV2.IntezerV2 import get_latest_result_command
```
Comment (yaakovpraisler) [inline Packs/Intezer/Integrations/IntezerV2/IntezerV2_test.py:16]: Please remove this if it's not necessary.

File: Packs/Intezer/ReleaseNotes/1_2_0.md (hunk 6 above)
Comment (yaakovpraisler) [inline Packs/Intezer/ReleaseNotes/1_2_0.md:14]: Please add a release notes entry for the updated docker image.
You can see a template example here - https://xsoar.pan.dev/docs/documentation/release-notes#docker-updates-example.

**Comment by yaakovpraisler** [comment]
If you want, you can give me permission and I will make the small changes.

File: Packs/Intezer/Integrations/IntezerV2/IntezerV2_test.py (hunk 11 above)
Comment (kfirstri) [reply Packs/Intezer/Integrations/IntezerV2/IntezerV2_test.py:16]: done

File: Packs/Intezer/ReleaseNotes/1_2_0.md (hunk 6 above)
Comment (kfirstri) [reply Packs/Intezer/ReleaseNotes/1_2_0.md]: :+1:

**Comment by kfirstri** [comment]
Hi @yaakovpraisler just added everything we talked about in the demo :)

**Comment by kfirstri** [comment]
the "Run Validations" failed because something failed in the demisto-sdk..

**Comment by yaakovpraisler** [comment]
If you can, please change the branch name to something else without the word master.
Anyway, please try to merge from master again, and let's see what happens.

**Comment by kfirstri** [comment]
closing this PR for the new PR :) 
https://github.com/demisto/content/pull/13751
@yaakovpraisler i changed the branch name
